public/video-ui/src/services/UploadsApi.js

```diff
@@ -51,49 +50,73 @@ function getCredentials(id, key) {
   });
 }
 
-function getS3(bucket, region, credentials) {
+/**
+ *
+ * @param region {string}
+ * @param credentials {any}
+ * @returns {S3}
+ */
+function getS3(region, credentials) {
   const { temporaryAccessId, temporarySecretKey, sessionToken } = credentials;
-  const awsCredentials = new AWS.Credentials(
-    temporaryAccessId,
-    temporarySecretKey,
-    sessionToken
-  );
-
-  return new AWS.S3({
-    apiVersion: '2006-03-01',
+
+  const awsCredentials = {
+    accessKeyId: temporaryAccessId,
+    secretAccessKey: temporarySecretKey,
+    sessionToken: sessionToken
+  };
+
+  return new S3({
     credentials: awsCredentials,
-    params: { Bucket: bucket },
+    requestHandler: XhrHttpHandler.create({
+      requestTimeout: 240_000
+    }),
     region: region,
     useAccelerateEndpoint: true
   });
 }
 
+/**
+ * Upload single part of file
+ *
+ * @param upload {Upload}
+ * @param part {typeof Upload['parts'][number]}
+ * @param file {File}
+ * @param progressFn {(completed: number) => any}
+ * @returns {Promise<unknown>}
+ */
 function uploadPart(upload, part, file, progressFn) {
   const slice = file.slice(part.start, part.end);
 
   return getCredentials(upload.id, part.key).then(credentials => {
     const s3 = getS3(
-      upload.metadata.bucket,
       upload.metadata.region,
       credentials
     );
 
-    const params = {
+    const request = slice.arrayBuffer().then(body => s3.putObject({
+      Bucket: upload.metadata.bucket,
       Key: part.key,
-      Body: slice,
-      ACL: 'private',
-      Metadata: { original: file.name }
-    };
-    const request = s3.upload(params, httpOptions);
-
-    request.on('httpUploadProgress', event => {
-      progressFn(part.start + event.loaded);
+      Metadata: { original: file.name },
+      Body: body
```

Comment:
Is there a behavioural change here in passing in the body rather than the `slice` variable above? Was that intentional?

Reply:
Ah right ok, makes sense. Small point and I know this was how it was before, but there could be a slightly more helpful variable name than slice here too.